### DISCUSSION TITLE
How to access generic python function across multiple workbooks with limited columns

### DISCUSSION THREAD

**Discussion by siddanthk** [tableau/TabPy#470] [Q&A] [answered]
Is there a possible way to execute a python function in multiple workbooks with each workbook having only some input arguments for the function. Something like keyword arguments or passing null columns if they are not present in the workbook.

ex:

def max(a,b,c):
if a:
return max(a)
elif b:
return min(b)
elif c:
return avg(c)

Workbook 1 has only columns 'a' and 'b', where as workbook 2 has columns 'b' and 'c'. I want to use same function in both the workbooks for calculating fields. Is there a way to handle this request apart from writing multiple functions (as i have 300+ workbooks)?

**Comment by 0golovatyi** [comment, marked as answer]
Answered your question here  - https://github.com/tableau/TabPy/issues/468
